src/Microsoft.OData.Core/UriParser/TypePromotionUtils.cs

```diff
@@ -493,8 +493,20 @@ internal static bool CanConvertTo(SingleValueNode sourceNodeOrNull, IEdmTypeRefe
                 return true;
             }
 
+			// We must support conversion from/to untyped.
+			if (sourceReference.IsUntyped() || targetReference.IsUntyped())
+			{
+				return true;
+			}
+
             if (targetReference.IsODataComplexTypeKind() || targetReference.IsODataEntityTypeKind())

```

Comment:
>targetReference.IsODataComplexTypeKind() || targetReference.IsODataEntityTypeKind() [](start = 16, length = 83)

maybe it's better to add an internal method named "IsODataStructuredTypeKind()" and use it in if clause?

Reply:
You should be able to use targetReference.IsStructured().

---
In reply to: [320888067](https://github.com/OData/odata.net/pull/1489#discussion_r320888067) [](ancestors = 320888067)